### PR TITLE
Add 'latest' parameter to -MSBuildVersion

### PR DESCRIPTION
**Pull request by goyzhang**
Add a "latest" parameter to -MSBuildVersion such that a user with an unwanted environment variable  PATH to an old tool can still specify using the highest version of MSBuild.
https://github.com/NuGet/Home/issues/8808

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs]: I don't think this does it.

If you look at line 502, this will still go into the path env var: https://github.com/NuGet/NuGet.Client/blob/0735174f6a7d01445d63a11df7cbc13d07e16353/src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs#L495-L506

File: src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs
Comment (zkat) [inline src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs]: Instead of doing it this way, I would rather you mirror the implementation in `GetMsBuildToolset` and either set `userVersion` to `string.Empty`, or just do `if (string.IsNullOrEmpty(userVersion) || string.Equals(userVersion, "latest", StringComparison.OrdinalIgnoreCase)` over here:
https://github.com/NuGet/NuGet.Client/blob/5564b076b3055a16b1765b3abd3f67025d484a28/src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs#L639-L645

I believe this will be much clearer. Your other change seems good, though!

**Review by zkat — COMMENTED** [review]
Just a small suggestion for readability (imo) and we're all good, I think :)

File: src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs
Comment (goyzhang) [reply src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs]: Done, thanks for the suggestion!

**Review by zkat — APPROVED** [review]
LGTM! Thanks for the quick turnaround :)

Just gonna wait for CI to complete to merge, and we're good.

**Review by nkolev92 — COMMENTED** [review]
The change looks good. 

Can you please add some tests similar to https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs.

**Comment by goyzhang** [comment]
> The change looks good.
> 
> Can you please add some tests similar to https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs.

OK. I took a look at the test class. It seems to focus on testing `GetMsBuildDirectoryInternal`. The `GetMsBuildToolset` is barely covered by the test. And it is hard to test `GetMsBuildToolset`. Besides, the EnvironmentVariable will be checked again in `GetMsBuildDirectoryInternal`. I will move the change into `GetMsBuildDirectoryInternal`.

**Comment by nkolev92** [comment]
> Besides, the EnvironmentVariable will be checked again in GetMsBuildDirectoryInternal

Feel free to use the abstraction for retrieving environment variables as necessary. Other parts of the same class already do it. It will allow you to test things better.

**Comment by goyzhang** [comment]
> > Besides, the EnvironmentVariable will be checked again in GetMsBuildDirectoryInternal
> 
> Feel free to use the abstraction for retrieving environment variables as necessary. Other parts of the same class already do it. It will allow you to test things better.

OK, thanks. How is the test in the last commit? Is it OK?

**Review by zkat — APPROVED** [review]
tests look good. Approved pending CI

**Comment by zkat** [comment]
Note: We're having CI issues right now, but we should be able to take this when they're cleared up :)

Thanks a lot for the patch!

**Comment by zkat** [comment]
WELL OOPS. LOL

**Comment by zkat** [comment]
see https://github.com/NuGet/NuGet.Client/pull/3370 instead >.<

**Pull request closed without merging**
